metatensor-torch/src/atomistic/system.cpp

```diff
@@ -214,31 +214,30 @@ std::vector<torch::Tensor> NeighborsAutograd::backward(
 
     auto positions_grad = torch::Tensor();
     if (positions.requires_grad()) {
-        positions_grad = torch::zeros_like(positions);
-
-        for (int64_t sample_i = 0; sample_i < samples.size(0); sample_i++) {
-            auto atom_i = samples[sample_i][0].item<int32_t>();
-            auto atom_j = samples[sample_i][1].item<int32_t>();
-
-            auto all = torch::indexing::Slice();
-            auto grad = distances_grad.index({sample_i, all, 0});
-            positions_grad.index({atom_i, all}) -= grad;
-            positions_grad.index({atom_j, all}) += grad;
-        }
+        auto positions_grad_neighbors = torch::zeros_like(positions);
+        positions_grad_neighbors = torch::index_add(
+            positions_grad_neighbors,
+            /*dim=*/0,
+            /*index=*/samples.index({torch::indexing::Slice(), 1}),
+            /*source=*/distances_grad.squeeze(-1)
+        );
+        auto positions_grad_centers = torch::zeros_like(positions);
```

Comment:
why a new allocation? Could this be done with the same tensor? (maybe using `alpha=-1`)

Reply:
I misinterpreted your request above. Yes, it can be done in many ways